Framework/Algorithms/src/CorrectTOFAxis.cpp

```diff
@@ -0,0 +1,503 @@
+#include "MantidAlgorithms/CorrectTOFAxis.h"
+
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidAPI/InstrumentValidator.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidKernel/ListValidator.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/Run.h"
+#include "MantidAPI/SpectrumInfo.h"
+#include "MantidAPI/WorkspaceProperty.h"
+#include "MantidAPI/WorkspaceUnitValidator.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/MandatoryValidator.h"
+#include "MantidKernel/PhysicalConstants.h"
+#include "MantidKernel/UnitConversion.h"
+
+namespace Mantid {
+namespace Algorithms {
+
+using Mantid::Kernel::Direction;
+using Mantid::API::WorkspaceProperty;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(CorrectTOFAxis)
+
+namespace {
+/** A private namespace holding some column names for FindEPP algorithm's output
+ *  table.
+ */
+namespace EPPTableLiterals {
+/// Title of the fit status column in EPP tables
+const static std::string FIT_STATUS_COLUMN("FitStatus");
+/// Title of the peak centre column in EPP tables
+const static std::string PEAK_CENTRE_COLUMN("PeakCentre");
+/// Tag for successfully fitted rows in EPP tables
+const static std::string FIT_STATUS_SUCCESS("success");
+} // namespace EPPTableLiterals
+
+/** A private namespace listing the different ways to index
+ *  spectra in Mantid.
+ */
+namespace IndexTypes {
+/// Tag for detector ids
+const static std::string DETECTOR_ID("Detector ID");
+/// Tag for spectrum numbers
+const static std::string SPECTRUM_NUMBER("Spectrum Number");
+/// Tag for workspace indices
+const static std::string WORKSPACE_INDEX("Workspace Index");
+} // namespace IndexTypes
+
+/** A private namespace listing the properties of CorrectTOFAxis.
+ */
+namespace PropertyNames {
+const static std::string ELASTIC_BIN_INDEX("ElasticBinIndex");
+const static std::string EPP_TABLE("EPPTable");
+const static std::string INCIDENT_ENERGY("IncidentEnergy");
```

Comment:
At least the two indirect instruments I have worked with (the backscattering spectrometers IN16b at ILL and SPHERES at MLZ) don't have any notion of TOF and thus there is no need to deal with the TOF axis at all. I don't know about other indirect instruments, though. The use cases I have planned for this algorithm include only direct geometry instruments but of course I cannot predict how other people might use this.

Anyway, for me this boils down to if I would like to be consistent with some algorithms I have written earlier. In hindsight I think should have used *EFixed* in them as well. I guess this is as good point as any to start using *EFixed*, so I'll make the change.